### ISSUE TITLE
hasCodeInUrl always returns the first value

### ISSUE THREAD

**Issue by andreas-sundstrom** [bjerkio/oidc-react#1074]
We have tried upgrading to latest version `3.4.0` in an application that uses hash-routing. When doing so login no longer works. We have identified the cause to be the following part of `hasCodeInUrl`:
```ts
  return (
    searchParams.has('code') ??
    searchParams.has('id_token') ??
    searchParams.has('session_state') ??
    hashParams.has('code') ??
    hashParams.has('id_token') ??
    hashParams.has('session_state')
  );
```
Since `searchParams.has('')` always returns true or false, never null, it will always default to that value, and never evaluate the other expressions.

We noticed that this changed in november 2023 but wasn't released until recently.

**Comment by andreassundstrom** [comment]
I have submitted PR #1075 for this issue.

**Comment by abebehailu** [comment]
+1

**Comment by simenandre** [comment]
Fixed by #1075 and released as https://github.com/bjerkio/oidc-react/releases/tag/v3.4.1. Thanks to @andreassundstrom for the PR 👍
